extensions/ConfirmEdit/ReCaptcha.php

```diff
@@ -23,43 +21,31 @@
 
 $wgAutoloadClasses['ReCaptcha'] = $dir . '/ReCaptcha.class.php';
 
-require_once( 'recaptchalib.php' );
-
-// Set these in LocalSettings.php
-# $wgReCaptchaPublicKey = '';
-# $wgReCaptchaPrivateKey = '';
-// For backwards compatibility
-# $recaptcha_public_key = '';
-# $recaptcha_private_key = '';
-
-/**
- * Sets the theme for ReCaptcha
- *
- * See http://code.google.com/apis/recaptcha/docs/customization.html
- */
-$wgReCaptchaTheme = 'white';
+$wgHooks['BeforePageDisplay'][] = 'efReCaptchaOnBeforePageDisplay';
 
 $wgExtensionFunctions[] = 'efReCaptcha';
 
 /**
  * Make sure the keys are defined.
+ *
+ * @throws Exception
  */
 function efReCaptcha() {
-	global $wgReCaptchaPublicKey, $wgReCaptchaPrivateKey;
-	global $recaptcha_public_key, $recaptcha_private_key;
-	global $wgServer;
+	$wg = F::app()->wg;
 
-	// Backwards compatibility
-	if ( $wgReCaptchaPublicKey == '' ) {
-		$wgReCaptchaPublicKey = $recaptcha_public_key;
-	}
-	if ( $wgReCaptchaPrivateKey == '' ) {
-		$wgReCaptchaPrivateKey = $recaptcha_private_key;
+	if ( $wg->ReCaptchaPublicKey == '' || $wg->ReCaptchaPrivateKey == '' ) {
+		throw new Exception( wfMessage( 'recaptcha-misconfigured' )->escaped() );
 	}
+}
 
-	if ( $wgReCaptchaPublicKey == '' || $wgReCaptchaPrivateKey == '' ) {
-		die ( 'You need to set $wgReCaptchaPrivateKey and $wgReCaptchaPublicKey in LocalSettings.php to ' .
-				"use the reCAPTCHA plugin. You can sign up for a key <a href='" .
-				htmlentities( recaptcha_get_signup_url ( str_replace( 'http://', '', $wgServer ), "mediawiki" ) ) . "'>here</a>." );
-	}
+/**
+ * @param OutputPage $out
+ * @param Skin $skin
+ *
+ * @return bool
+ */
+function efReCaptchaOnBeforePageDisplay( OutputPage &$out, Skin &$skin ) {
```

Comment:
This is a hook handler for `BeforePageDisplay` which is called with `OutputPage &$out` and `Skin &$skin`.  We need the full signature to not get errors.